extension/ts/content/extensions/version/version.ts

```diff
@@ -0,0 +1,37 @@
+interface Version {
+	major: number;
+	minor: number;
+	revision: number;
+}
+
+const versionCompare = (comparator: (a: number, b: number) => boolean) => (versionA: Version, versionB: Version) => {
+	if (comparator(versionA.major, versionB.major)) {
+		return true;
+	} else if (comparator(versionB.major, versionA.major)) {
+		return false;
+	} else if (comparator(versionA.minor, versionB.minor)) {
+		return true;
+	} else if (comparator(versionB.minor, versionA.minor)) {
+		return false;
+	} else if (comparator(versionA.revision, versionB.revision)) {
+		return true;
+	} else if (comparator(versionB.revision, versionA.revision)) {
+		return false;
+	} else {
+		return false;
+	}
+};
```

Comment:
i don't have any issues with this, and i'm sure it's good for readability but it kinda reminds me of this lol

https://twitter.com/ctrlshifti/status/1288745146759000064

Reply:
lol this is what it used to look like
https://github.com/braxtonhall/library-thing/blob/b482ff6554b14a9334a7ecbb0330e55883ab0d01/extension/ts/content/extensions/version.ts#L45-L58

there _must_ be a better way